[Core] Add a counter for installed user addins

mrward opened the pull request:
Reports the addin id and version for all installed and enabled user
addins.

Fixes VSTS #712478 - Add Telemetry to identify extensions installed

Currently this does not handle addins being disabled/enabled or installed/uninstalled.

The AddinManager.AddinUnloaded event can be used for disabling and uninstalling.

However AddinManager.AddinLoaded fires on startup or when an addin is lazily loaded. So it does not necessarily indicate an addin is being installed or enabled. It seems like we would need the AddinEngine's ActivateAddin to fire an event. That seems to be called by the AddinDatabase on installing a new addin.

Another possibility would be to cache the list of user installed addins and every time the AddinLoaded event is fired the AddinManager's Registry is checked to see if there is a difference. Although this would mean using memory just for a counter that might not be used during an IDE session.

Or not as good, we could send an event on shutdown listing the installed addins at the end of the IDE session.

mrward committed:
[Core] Add a counter for installed user addins

Reports the addin id and version for all installed and enabled user
addins.

Fixes VSTS #712478 - Add Telemetry to identify extensions installed

Therzok commented on main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:
```diff
@@ -154,7 +155,26 @@ public static void Initialize (bool updateAddinRegistry)
+
+			foreach (Addin addin in AddinManager.Registry.GetModules (AddinSearchFlags.IncludeAddins | AddinSearchFlags.LatestVersionsOnly)) {
+				if (addin.IsUserAddin && addin.Enabled) {
+					string id = Addin.GetFullId (addin.Namespace, addin.LocalId, null);
```
I think this counts as PII. Should we hash them and have a list of well known IDs?

mrward replied:
Doing the PII hashing in md-addins.

mrward commented:
Another idea here might be to check the user installed adds before and after the addin manager dialog is displayed (AddinManagerWindow) and send out a counter if they have changed. Seems to be multiple places where the addin manager dialog can be triggered - main menu, check for updates (AddinsUpdateHandler), opening an .mpack file.

mrward committed:
[Ide] Report changes to user enabled addins

After the Extension Manager dialog is closed a check is made to see if
the user enabled addins has changed and if so this is reported with
an instrumentation counter.

mrward commented:
Changed this around a bit. Moved the counter to MonoDevelop.Ide. Also handle addins being installed/enabled/uninstalled/disabled from the Extension Manager.

Therzok commented on main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Updater/AddinsUpdateHandler.cs:
```diff
@@ -123,6 +128,68 @@ public static void HideAlert ()
+				foreach (Addin addin in GetEnabledUserAddins ()) {
+					addins.Add (addin.Id);
+				}
+				addins.Sort ();
```
Do we need to sort it?

mrward replied:
Sorting it for the comparison. It does not look like they are sorted when returned from the AddinManager. Was assuming we could not rely on the order returned from the AddinManager when looking for changes.

mrward commented:
Implementing a different approach.
